arraycontext/container/traversal.py

```diff
@@ -583,7 +583,10 @@ def thaw(ary: ArrayOrContainerT, actx: ArrayContext) -> ArrayOrContainerT:
 
 # {{{ flatten / unflatten
 
-def flatten(ary: ArrayOrContainerT, actx: ArrayContext) -> Any:
+def flatten(
+        ary: ArrayOrContainerT, actx: ArrayContext, *,
+        leaf_class: Optional[type] = None,
```

Comment:
I'm not sure `leaf_class` is a great name. `flatten_at_class`?

Reply:
@inducer Mostly went with what was used in `rec_map_array_container` and friends in #128, since I would expect them to all use the same argument name. Would you prefer that they each have their own names like this?

Reply:
Make an issue of it?

Reply:
Ah :shrug:. Terminology shmerminology.